lib/queue.js

```diff
@@ -680,6 +686,7 @@ Queue.prototype.processJob = function(job){
     _this.processing--;
     var error = err.cause || err; //Handle explicit rejection
     return job.moveToFailed(err)
+      .then(clearTimer)
```

Comment:
How so? The worst case is that it gets renewed after moving to failed, but then immediately released.